internal/services/git/git_test.go

```diff
@@ -110,6 +110,14 @@ func TestGetCommitAuthor(t *testing.T) {
 	t.Run("Should return a new service", func(t *testing.T) {
 		assert.NotEmpty(t, New(&config.Config{}))
 	})
+
+	t.Run("Should not return git diff", func(t *testing.T) {
+		bytes, err := service.executeCMD("5", "README.md")
+		
+		assert.NoError(t, err)
+		assert.NotEmpty(t, bytes)
+		assert.NotContains(t, string(bytes), "diff --git")
+	})
```

Comment:
I think that would be better to use the public API instead private method. Private methods change constantly , public API not. What do you think about create a `internal/services/git/testdata` directory with a file with the following content:
```go
package foo
const something = '{"foo": "bar"}'
```
And them use this file as a input on `CommitAuthor` method

Reply:
Calling only the public API, the test would not vary much from the previous ones and I believe it would even be repetitive. By calling the private method I can validate exactly the error scenario that happened, and make a more accurate test scenario.

If the method changes and breaks the test, whoever is making the changes will be aware  this scenario and will make the necessary changes to the new implementation. Which I believe is exactly the purpose of the test.

About creating the test directory. Is it really necessary? I don't see any problem using README.md in this scenario, because even if the author changes, we're not validating the data, so it wouldn't impact the test. One problem would be to remove the file, but I don't see a future that our project doesn't have a README.md at the root. Sounds a little unnecessary to me.

Reply:
The bug was caused using the public API, so testing the public API with other kinds of inputs will catch this bug and will not be repetitive. I don't think that is a good strategy change a private method and break the tests, if we improve the internal structures without breaking the public API the tests should still passing. If we will test each private method the tests will be too hard to maintain.

The bug occur because the diff contains a `}` witch will broke the parsing, the README.md does not have this `}`, so this is the reason why I suggest creating this file (and we don't need to remove it).

Reply:
Also, check this proposal for Go 2: https://github.com/golang/go/issues/29258

Reply:
I had not seen the function that removes the diff, with the new changes it is no longer necessary. Now any diff information will break the parse, it is not necessary to add the new file.

About the tests, I'm not saying to test all private methods, but I believe that in some scenarios we can create more accurate tests about the error.

Reply:
I think that the testcase should use the data that cause the bug, the README.md not cause this bug.

I understand your point but I still disagree about testing private methods but I'll resolve this and let for further discussion.

Reply:
With the changes the bug will happen if the output contains the diff information, not being unique to the character `}`. So any files that return extra information will break parse, including README.md.